### PR TITLE
[feature](Nereids): pushdown COUNT(*) through join

### PR DESCRIPTION
**Pull request by jackwener**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by jackwener** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 44.96 seconds
 stream load tsv:          512 seconds loaded 74807831229 Bytes, about 139 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.3 seconds inserted 10000000 Rows, about 341K ops/s
 storage size: 17162068132 Bytes

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/PushdownCountThroughJoin.java [hunk 1]
```diff
@@ -43,8 +43,28 @@
...
+ *  (x)
+ *  ->
+ *  aggregate: Sum( cnt * cntStar )
+ *  |
+ *  join
+ *  |   \
+ *  |    aggregate: count(*) as cntStar
+ *  aggregate: count(x) as cnt
+ *  </pre>
+ * Notice: when Count(*) exists, group by mustn't be empty.
```
Comment (englefly) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/PushdownCountThroughJoin.java:67]: please update this comments as we discussed

**Review by englefly — APPROVED** [review]

**Review by englefly — CHANGES_REQUESTED** [review]
How about count(1) ?

**Review by starocean999 — APPROVED** [review]

**Review by englefly — APPROVED** [review]

**Pull request merged**